### PR TITLE
added checkers to make sure no network tap actions can be taken if th…

### PR DESCRIPTION
**Pull request by Jeff-Wang93**
…e network tap core tool isn't installed, built, and running

### CODE REVIEW

**Comment by Jeff-Wang93** [comment]
issue #803

File: vent/api/actions.py [hunk 1]
```diff
@@ -1121,6 +1121,83 @@ def enable(self,
...
+        Returns:
+            A tuple of success status, and a tuple containing:
+            bool describing if plugin manifest exists,
+            bool describing if tool is built,
+            bool describing if tool is running.
+            eg: (True, (True, True, False))
+        """
+        status = (True, None)
+        try:
+            # problem is that currently does not throw error if plugin_manifest
```
Comment (cglewis) [inline vent/api/actions.py]: is this still a problem with these changes, or should this comment be read as this try/except is here to throw an error if the plugin_manifest doesn't exist?

File: vent/api/actions.py [hunk 2]
```diff
@@ -1121,6 +1121,83 @@ def enable(self,
...
+
+            # this means the core tool isn't built
+            elif tool_status[1] == 'no':
+                output = "Please build core tool " + str(tool_name)
+
+            # this means the core tool isn't running
+            elif tool_status[2] == 'no':
+                output = "Please start core tool " + str(tool_name)
+            status = (True, output)
+        except Exception as e:
```
Comment (cglewis) [inline vent/api/actions.py]: add pragma marker

File: vent/api/actions.py [hunk 3]
```diff
@@ -1121,6 +1121,83 @@ def enable(self,
...
+            self.logger.info("Start: tool_status_checker")
+            manifest = Template(self.p_helper.manifest)
+            for section in manifest.sections()[1]:
+                if manifest.option(section, 'name')[1] == tool_name:
+                    status_tup = (True, manifest.option(section, 'built')[1],
+                                  manifest.option(section, 'running')[1])
+                    status = (True, status_tup)
+                    break
+
+        except Exception as e:
```
Comment (cglewis) [inline vent/api/actions.py]: add pragma marker

File: vent/menus/main.py [hunk 4]
```diff
@@ -415,38 +415,85 @@ def system_commands(self, action):
             # add notify_cancel_ok popup once implemented
             pass
         elif action == "ntapcreate":
-            form_args = {'color': 'CONTROL',
-                         'name': 'Network Tap Interface Create' + "\t"*6 +
-                                 '^T to toggle main'}
-            self.add_form(CreateNTap, "Network Tap Create", form_args)
+            output = self.api_action.tool_status_output('network_tap')
```
Comment (cglewis) [inline vent/menus/main.py]: a lot of repetition inside the `elif`s, can't some of those be moved out and only exist once?

File: vent/api/actions.py (hunk 1 above)
Comment (Jeff-Wang93) [reply vent/api/actions.py]: The ladder. It's checking to make sure that the manifest exists. The comment was me keeping track of why the function wasn't working before.

File: vent/menus/main.py (hunk 4 above)
Comment (Jeff-Wang93) [reply vent/menus/main.py]: You're right. I'll try and reduce the repetition.

**Pull request merged**